src/styles/blocks/catalog.scss

```diff
@@ -0,0 +1,19 @@
+.catalog {
+  display: grid;
+  justify-content: center;
+  @include catalogGridTemplateColumns(4);
+  gap: $catalog-gap;
+  padding: $catalog-paddingY $catalog-paddingX;
+
+  @media (max-width: 1023px) {
+    @include catalogGridTemplateColumns(3);
+  }
+
+  @media (max-width: 767px) {
+    @include catalogGridTemplateColumns(2);
+  }
+
+  @media (max-width: 487px) {
+    @include catalogGridTemplateColumns(1);
+  }
```

Comment:
You can make a loop to create all media queries. Just create a correctly mapped list to iterate.